pkgs/build-support/setup-hooks/patch-ant-buildfiles.sh

```diff
@@ -0,0 +1,30 @@
+
+
+# Sets the `modificationtime` attribute on the <jar> and <war> ant tasks
+# This is done to make the final archives deterministic
+patchAntBuildfile() {
+    local buildfile="$1"
+    # using `local-name()` ignores the `xmlns` attribute in a parent node
+    @xmlstarlet@ ed --inplace \
+        --var T '//*[local-name()="jar" or local-name()="war"]' \
+        --update '$T/@modificationtime' -v 0 \
+        --insert '$T[not(@modificationtime)]' -t attr -n 'modificationtime' -v 0 \
+        "$buildfile"
+}
+
+patchAntBuildfilesHook() {
+    find "$NIX_BUILD_TOP" -name '*.xml' -type f -writable -print0 |
+        while IFS= read -r -d '' file; do
+            # check if the xml file has any <target> nodes inside a top-level <project> node
+            # this is done to only patch ant buildfiles, not just any xml file
+            # using `local-name()` ignores the `xmlns` attribute in the <project> node
+            if @xmlstarlet@ sel -Q -t -c '/*[local-name()="project"]/*[local-name()="target"]' "$file"; then
```

Comment:
Quote `@xmlstarlet@`.

Reply:
Again, use `lib.escapeShellArg` when setting replacements and it will be both more correct and you won't need to think about quoting at every use-site.

Reply:
We could also just use `propagatedBuildInputs`, though that would spill the implementation (the fact that this uses `xmlstarlet`) out into the build environment.
I went ahead and used `lib.escapeShellArg`